js/compiler.ts

```diff
@@ -167,6 +186,57 @@ export class DenoCompiler implements ts.LanguageServiceHost {
   // testing
   private _window = window;
 
+  /**
+   * Drain the instantation queue, retrieving the arguments for the module
+   * factory and calling the module's factory.
+   */
+  private _drainInstantateQueue(): void {
+    this._log(
+      "compiler._drainInstantateQueue",
+      this._instantiateQueue.map(metaData => metaData.fileName)
+    );
+    let moduleMetaData: ModuleMetaData | undefined;
+    while ((moduleMetaData = this._instantiateQueue.shift())) {
+      assert(
+        moduleMetaData.factory != null,
+        "Cannot instantiate module without factory."
+      );
+      assert(!moduleMetaData.hasRun, "Module has already been instantiated.");
```

Comment:
I would prefer you use the more descriptive `moduleMetaData.hasRun != null`

Reply:
We would `=== false` here then, or `== null` since we are ensuring it hasn't run.